### PR TITLE
Allow remote configuration of the Report Issue / Help module

### LINKED ISSUE #2054: "Report a Problem" screen tweaks
- rename to "Help"?
- add two more cards – maybe how to report something to the helpdesk, and to facilities?
- visual tweakses

suggestions welcome

### PR DESCRIPTION
**Pull request by hawkrives**
<details><summary>screenshots</summary>

Note | Screenshot
--- | ---
Not scrolled | screen shot 2018-01-13 at 11 48 01 pm
Scrolled | screen shot 2018-01-13 at 11 48 04 pm
Disabled tool, but not hidden | screen shot 2018-01-13 at 11 54 00 pm
A single button disabled | screen shot 2018-01-13 at 11 54 34 pm
Disabled with custom message | screen shot 2018-01-13 at 11 59 07 pm
</details>

Whee!

I think this feature is just about ready to go.

I've built out the ability to configure simpler "tools" like the Helpdesk (open url, send email, call phone) completely from the remote json file, while we can still enable/disable/hide/edit the title/description of the wifi reporter.

We can also configure the order of the rendered tools by reordering the YAML files that make up the data file.

I think the current files I've included to configure the app are fairly explanatory, but I will note that the Facilities entry includes an idea-based remote form configuration. For now, I'm just opening the (hideous) facilities form in an SFWebView – I don't want to add an http exception, and I don't want to build a native form right now. If you want to report it badly enough, you'll suffer through the form lol.

We can disable individual buttons on the tools, too. IDK why.

EDIT: We can also restrict tools to only certain versions of the app; there's a key that takes a server-compatible version query, like `>=2.5.0 <2.6`, so we could build several tools in case there's something backwards incompatible in here somewhere, and only the one that worked on your app version would show.

I put a message in the prose for the Helpdesk to explain that you need to be on-campus for the website link to work. I don't really want to do anything more at this time.

I tweaked the styles of the cards again; I removed the big bold Title style, because I remembered that there was a `header={}` prop for Card itself. If we don't like that style, we should think about improving it everywhere, not just changing it here. I also removed the left/right edge margin, and fixed the bottom margin on the list to match the top margin and the space between the cards.

---

Should we add a button in here for getting help with the app, too? Just a random late-night thought.

TL;DR: After this PR, this feature is no longer a shipping hold.

Closes #2054. We can open a new issue if we need more tweaks. 😸

### CODE REVIEW

**Comment by hawkrives** [comment]
The Flow errors are because this depends on the `<Card/>` tweaks in #2125

**Comment by hawkrives** [comment]
I think that some of the communications-esque methods in the new tools file could be pulled out into something to solve https://github.com/StoDevX/AAO-React-Native/issues/1799.

**Comment by rye** [comment]
I just approved #2125, I'll take a look once the thing rebuilds after that gets merged.

File: data/_schemas/help.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  enabled: {type: boolean}
+  hidden: {type: boolean}
+  message: {type: string}
+  versionRange: {type: string}
+  title: {type: string}
+  body: {type: string}
+  buttons:
+    type: array
+    items:
+      # TODO: convert this to "enum" instead of "oneOf"
```
Comment (rye) [inline data/_schemas/help.yaml:16]: Does this TODO need resolving in this PR?

**Review by rye — APPROVED** [review]
Looks pretty good to me.  I have no idea how well/if it will work in the wild, but we'll see, now won't we?  I'm fine with having this go into tomorrow's nightly.

File: data/_schemas/help.yaml (hunk 1 above)
Comment (hawkrives) [reply data/_schemas/help.yaml:16]: … no

I've tested that `oneOf` properly rejects items if they don't match. I believe that `enum` would provide nicer errors, but I couldn't get it to work.

So, no, this doesn't need to be resolved for a merge.
